src/TextInputWidget.cc

```diff
@@ -170,6 +226,13 @@ FilteredTextEdit::textChanged()
         working_history_[history_index_] = toPlainText();
 }
 
+void
+FilteredTextEdit::receiveImage(const QByteArray &img, const QString &img_name)
+{
+        QSharedPointer<QBuffer> buffer{new QBuffer{const_cast<QByteArray *>(&img), this}};
```

Comment:
In addition to being const-unsafe (it is almost never correct to use `const_cast`), this exposes a pointer to the `imageData_` member of `PreviewImageOverlay`. What happens if the user pastes in another image while the emitted `QBuffer` is still in use somewhere?

Instead, create a `QBuffer` that owns its own storage using its default constructor, and initialize it by calling [`QBuffer::setData`](https://doc.qt.io/qt-5/qbuffer.html#setData).